onnxscript/function_libs/torch_aten/ops/core.py

```diff
@@ -2174,10 +2176,19 @@ def aten_index_reduce(
     raise NotImplementedError()
 
 
-def aten_index_select(self: TensorType, dim: int, index: TensorType) -> TensorType:
+@torch_op("aten::index_select")
+def aten_index_select(self: TTensor, dim: int, index: TInt) -> TTensor:
     # index_select(Tensor self, int dim, Tensor index) -> Tensor
 
-    raise NotImplementedError()
+    if op.Size(op.Shape(index)) == 0:
```

Comment:
Otherwise, is `index` expected to be 1-dimensional? Or, can it have any other rank?

Reply:
Yes, it should be 1d I think.

Reply:
should I do `op.Reshape(index, [-1])` instead?

Reply:
Yes, if we expect it to be 0d or 1d, I think reshape, as you mention, is a better option.

Reply:
Done. thanks!